angular2-twitter/public/src/app/data-creator.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class DataCreatorService {
+
+  constructor() { }
+
+  createData(sortedData: Object, colors: string[]) {
+    let i = 0;
+    let data = {
+      bar: [],
+      map: [],
+      linear: []
+    }
+    Object.entries(sortedData).forEach(([key, value]) => {
+      data.bar = this.createBarData(data.bar, key, value, colors[i]);
+      data.map = this.createMapData(data.map, key, value, colors[i]);
+      data.linear = this.createLinearData(data.linear, key, value, colors[i]);
+      i++;
+    });
+    return data;
```

Comment:
co do tej funkcji to najlepszym sposobem byłoby tu zastosowanie .reduce nie .forEach
```js
Object
  .entries(sortedData)
  .reduce((acc, [key, value]) => {
    acc.bar = acc.bar.concat(this.createBarData({ key, value, color: colors[acc.i] }));
    ...
    acc.i += 1;
    return acc
  }, { bar: [], map: [], linear: [] })

Reply:
https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Array/Reduce